components/speedreader/speedreader_whitelist.cc

```diff
@@ -40,24 +65,23 @@ void SpeedreaderWhitelist::OnComponentReady(const std::string& component_id,
                                             const std::string& manifest) {
   base::PostTaskAndReplyWithResult(
       FROM_HERE, {base::ThreadPool(), base::MayBlock()},
-      base::BindOnce(&brave_component_updater::LoadDATFileData<adblock::Engine>,
-                     install_dir.Append(kDatFileName)),
+      base::BindOnce(
+          &brave_component_updater::LoadDATFileData<speedreader::SpeedReader>,
+          install_dir.Append(kDatFileVersion).Append(kDatFileName)),
       base::BindOnce(&SpeedreaderWhitelist::OnGetDATFileData,
                      weak_factory_.GetWeakPtr()));
 }
 
 bool SpeedreaderWhitelist::IsWhitelisted(const GURL& url) {
-  bool explicit_cancel;
-  bool saved_from_exception;
-  std::string redirect;
+  return speedreader_->IsReadableURL(url.spec());
+}
 
-  return engine_->matches(url.spec(), url.host(), url.host(),
-                          false, "sub_frame", &explicit_cancel,
-                          &saved_from_exception, &redirect);
+std::unique_ptr<Rewriter> SpeedreaderWhitelist::MakeRewriter(const GURL& url) {
```

Comment:
for followup - seems like this method belongs in the SpeedreaderService along with IsWhitelisted/IsReadableURL